src/odemis/driver/symphotime.py

```diff
@@ -777,18 +790,13 @@ def _actOnMessage(self,decoded_msg):
                 # Sent and acknowledges the error state clearing.
                 # Change the state back to RUNNING
                 self.state._set_value(model.ST_RUNNING, force_write=True)
-                    
+
             elif err_status == PQ_ERRCODE_MEASUREMENT_READY:
                 # Measurement is complete
                 if not self._measurement_stopped.is_set():
                     logging.info("Acquisition completed successfully.")
                     self._measurement_stopped.set()
-                    # update metadata and send data array to subscribers
-                    md = self._metadata.copy()
-                    md[model.MD_DWELL_TIME] = self.scanner.dwellTime.value
-                    md[model.MD_ACQ_DATE] = self._tStart
-                    da = model.DataArray([[0]], md)
-                    self.data.notify(da)
+                    self._notifySubscribers()
```

Comment:
It feels like a pity. You've moved that code to a separate function, but then you added an almost identical code for the detector_live. How about making that new function a little more generic so it can be used in both cases? => pass the DataFlow and content of DataArray in the arguments?